### PR TITLE
ROX-14527: Install via operator

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

This PR changes OpenShift e2e tests to deploy stackrox using the operator:
- Installs the operator with `make -C operator kuttl deploy-via-olm`.
- Creates Central and SecuredCluster CRs appropriate for e2e tests.

PR level test runs are pretty [stable](https://prow.ci.openshift.org/pr-history/?org=stackrox&repo=stackrox&pr=4557) though one issue arose that might hold this PR (or require a test skip): [ROX-14666](https://issues.redhat.com/browse/ROX-14666).

All level test runs are a WIP.

Other changes:
- OSCI OpenShift hive clusters need an extra node to handle postgres.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

- [x] `all` tests
- [x] release builds

### CODE REVIEW

**Comment by gavin-stackrox** [comment]
@porridge I'd appreciate your operator wisdom for this PR. Thanks.

**Review by kovayur — COMMENTED** [review]
LGTM, waiting for a green build.

File: scripts/ci/openshift.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+# A collection of OpenShift related reusable bash functions for CI
+
+SCRIPTS_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../.. && pwd)"
+source "$SCRIPTS_ROOT/scripts/lib.sh"
+
+set -euo pipefail
+
+scale_worker_nodes() {
```
Comment (SimonBaeumer) [inline scripts/ci/openshift.sh:10]: Why is it necessary to scale the worker nodes in a QA environment?

File: scripts/ci/openshift.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    current_replicas="$(oc -n openshift-machine-api get machineset "$first_worker_machine_set" -o json | jq -r '.spec.replicas')"
+    new_machineset_count=$((current_replicas + increment))
+    oc -n openshift-machine-api scale machineset "$first_worker_machine_set" --replicas="$new_machineset_count"
+
+    expected_count=$((original_count + increment))
+    # note: timeout of this loop is handled by the calling context
+    while [[ "$(get_running_worker_count)" != "$expected_count" ]]; do
+        info "Current machine state does not have the running count we desire ($expected_count)"
+        oc  -n openshift-machine-api get machines
+        sleep 60
```
Comment (SimonBaeumer) [inline scripts/ci/openshift.sh:36]: `sleep 60` seems long, can this be reduced?

**Review by SimonBaeumer — APPROVED** [review]
I really like that e2e tests are executed based on the operator installation. Thanks for this change @gavin-stackrox!

File: scripts/ci/openshift.sh (hunk 1 above)
Comment (gavin-stackrox) [reply scripts/ci/openshift.sh:10]: > Why is it necessary to scale the worker nodes in a QA environment?

This is for postgres i.e. the postgres overhead + compliance needs more CPU.

**Comment by gavin-stackrox** [comment]
I'm going to go ahead and merge. The failures in `openshift-penultimate-qa-e2e-tests` are not related to this PR and fail on current master because collector cannot find the drivers for `collector-ebpf-4.18.0-305.76.1.el8_4.x86_64.o` and that unfortunately surfaces as an overall test timeout.

**Pull request merged**